packages/extension/src/serviceWorker/index.ts

```diff
@@ -17,790 +17,373 @@
-    }
-  } catch (error) {
-    //Fail silently
-  }
-});
+      targets = await chrome.debugger.getTargets();
 
-/**
- * Fires when a window is removed (closed).
- * @see https://developer.chrome.com/docs/extensions/reference/api/windows#event-onRemoved
- */
-chrome.windows.onRemoved.addListener((windowId) => {
-  syncCookieStore?.removeWindowData(windowId);
-});
-
-/**
- * Fires when the extension is first installed,
- * when clicked on the extension refresh button from chrome://extensions/
- * when the extension is updated to a new version,
- * when Chrome is updated to a new version.
- * @see https://developer.chrome.com/docs/extensions/reference/api/runtime#event-onInstalled
- * @todo Shouldn't have to reinstall the extension.
- */
-chrome.runtime.onInstalled.addListener(async (details) => {
-  syncCookieStore = new SynchnorousCookieStore();
-  syncCookieStore?.clear();
-
-  setInterval(async () => {
-    await chrome.storage.session.get();
-  }, 20000);
-
-  if (details.reason === 'install') {
-    await chrome.storage.sync.clear();
-    await chrome.storage.sync.set({
-      allowedNumberOfTabs: 'single',
-      isUsingCDP: false,
-    });
-  }
-
-  if (details.reason === 'update') {
-    await chrome.storage.local.clear();
-    const preSetSettings = await chrome.storage.sync.get();
-    tabMode = preSetSettings?.allowedNumberOfTabs ?? 'single';
-    globalIsUsingCDP = preSetSettings?.isUsingCDP ?? false;
-    if (
-      preSetSettings?.allowedNumberOfTabs &&
-      Object.keys(preSetSettings).includes('isUsingCDP')
-    ) {
-      return;
-    }
-
-    await chrome.storage.sync.clear();
-    await chrome.storage.sync.set({
-      allowedNumberOfTabs: 'single',
-      isUsingCDP: false,
-    });
-  }
-});
-
-/**
- * Fires whenever debugging target issues instrumentation event.
- * @see https://developer.chrome.com/docs/extensions/reference/api/debugger
- */
-// eslint-disable-next-line complexity
-chrome.debugger.onEvent.addListener((source, method, params) => {
-  if (!ALLOWED_EVENTS.includes(method) || !source?.tabId || !params) {
-    return;
-  }
-
-  const url = syncCookieStore?.getTabUrl(source?.tabId);
-
-  const tabId = source.tabId.toString();
-
-  if (tabMode !== 'unlimited' && tabToRead !== tabId) {
-    return;
-  }
-
-  if (method === 'Network.responseReceived') {
-    const request = params as Protocol.Network.ResponseReceivedEvent;
-
-    // To get domain from the request URL if not given in the cookie line.
-    if (!requestIdToCDPURLMapping[tabId]) {
-      requestIdToCDPURLMapping[tabId] = {
-        [request.requestId]: request?.response.url,
-      };
-    } else {
-      requestIdToCDPURLMapping[tabId] = {
-        ...requestIdToCDPURLMapping[tabId],
-        [request.requestId]: request?.response.url,
-      };
-    }
-  }
+      await Promise.all(
+        targets.map(async ({ id, url }) => {
+          if (url.startsWith('http')) {
+            await attachCDP({ targetId: id });
```

Comment:
Are we attaching CDP to all target on all `ALLOWED_EVENTS` every time? Can it have any side effects when the same target is attached CDP more than once, like it happens with event listeners in JavaScript?

Reply:
If debugger is again attached to a target it errors out with an error saying `Debugger already attached to targetId: xyz`.
<img width="1164" alt="Screenshot 2024-05-29 at 10 57 00" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/bab3ee7b-bc1e-4d36-82f6-b6e054d709d7">

Reply:
Can you add a comment for it.